tests/src/test_cases/test_textarea.c

```diff
@@ -0,0 +1,85 @@
+#if LV_BUILD_TEST
+#include "../lvgl.h"
+
+#include "unity/unity.h"
+
+static lv_obj_t * active_screen = NULL;
+static lv_obj_t * textarea = NULL;
+
+void setUp(void)
+{
+    active_screen = lv_scr_act();
+    textarea = lv_textarea_create(active_screen);
+}
+
+void tearDown(void)
+{
+    /* Function run after every test */
+}
+
+void test_textarea_should_have_valid_documented_defualt_values(void)
+{
+    TEST_ASSERT(lv_textarea_get_cursor_click_pos(textarea));
+    TEST_ASSERT_EQUAL(0U, lv_textarea_get_one_line(textarea));
+    /* No placeholder text should be set on widget creation */
+    TEST_ASSERT_EQUAL_STRING("", lv_textarea_get_placeholder_text(textarea));
+}
+
+/* When in password mode the lv_textarea_get_text function returns
+ * the actual text, not the bullet characters. */
+void test_textarea_should_return_actual_text_when_password_mode_is_enabled(void)
+{
+    const char * text = "Hello LVGL!";
+
+    lv_textarea_add_text(textarea, text);
+    lv_textarea_set_password_mode(textarea, true);
+
+    TEST_ASSERT_TRUE(lv_textarea_get_password_mode(textarea));
+    TEST_ASSERT_EQUAL_STRING(text, lv_textarea_get_text(textarea));
+}
+
+void test_textarea_should_update_label_style_with_one_line_enabled(void)
+{
+    lv_textarea_add_text(textarea, "Hi");
+    lv_textarea_set_one_line(textarea, true);
+
+    TEST_ASSERT(lv_textarea_get_one_line(textarea));
+    // TEST_ASSERT_EQUAL_UINT16(LV_SIZE_CONTENT, lv_obj_get_width(ta->label));
```

Comment:
We can test the height to be: font's line height + top pad + bottom pad.

Reply:
Hi, this test was meant to test the changes done in 3e66581ba5ea2b21646d9b59836bcdad18788f1d, particularly this code in `lv_textarea_set_one_line`

```c
    lv_coord_t width = en ? LV_SIZE_CONTENT : lv_pct(100);
    lv_coord_t min_width_value = en ? lv_pct(100) : 0;

    lv_obj_set_width(ta->label, width);
    lv_obj_set_style_min_width(ta->label, min_width_value, 0);
```

Reply:
> AFAIK the label width should be LV_SIZE_CONTENT, isn't?

Yes, and it can be tested with `font's line height + top pad + bottom pad`. Or have I misunderstood something?

Reply:
No, I think I get it now, I will update this particular assert 😸

Reply:
Does 505da92 looks ok?